### ISSUE TITLE
Page Numbering with Macdown #enchancement

### ISSUE THREAD

**Issue by redconnection** [uranusjr/macdown#547]
Hi,

I'm really missing the option of numbering pages in Macdown. I understand that markdown has no concept of page numbers, however, it would be great to have it via:

1. PDF rendering engine if supports adding numbered pages
2. Possibility of adding a footer div to use with custom CSS. 

Thanks

**Comment by jkulesza** [comment]
I disagree with this enhancement request.  Markdown is intended to be a lightweight affair, both in terms of handling and output.  If you require "document" features, I would propose exploring other technologies (e.g., LaTeX, IPython and its notebooks, Mathematica notebooks).

I'm very excited to see how active the development is behind MacDown.  However, I'm becoming concerned with feature bloat.

**Comment by redconnection** [comment]
Truth is, if webkit would support "css3-page" properly, we wouldn't even have this conversation. However, looking at the age of this [request](https://bugs.webkit.org/show_bug.cgi?id=15548) (filed in 2007) i doubt that we will see this done. 

I digress, will use inline HTML. It sort of defeats the purpose of having content a styling separated though.  

Appreciate any better ideas.

**Comment by uranusjr** [comment]
Honestly there really isn’t anything practical I can do to support this. The best solution would be CSS, but as of 2016 no support is coming anywhere. Other solutions would be extremely hacky at best. This really is a very fundamental problem is HTML, and inherently Markdown.

For now I recommend using one of the following solutions:

1. Learn LaTeX.
2. If plain LaTeX is too difficult and/or too tedious to learn, restructuredText is a good alternative that is similar to Markdown. It supports page numbering, headers, and footers. You can generate a PDF either with rst2pdf or Pandoc.
3. Pandoc actually supports Markdown, so if you already have a Markdown document, you can first convert it to restructureText or LaTeX, add page numbering with minimal modification, and then outputs PDF.

I’m leaving this open to track possible future development in the CSS spec and WebKit implementation, but nothing will happen in MacDown before those are in place.

**Comment by Jmuccigr** [comment]
Is it not possible to have the page numbers added in the printing? Several apps do that.

**Comment by uranusjr** [comment]
Not with pure CSS. It is possible to add page numbers to printed documents (and PDF if you use the OS X printing system), but it would not be a pretty affair, and is (as @jkulesza mentioned) not quite in-line with a lightweight Markdown editor.

**Comment by Jmuccigr** [comment]
I don't see the problem with letting the print process add the page numbers (however that works). I use Macdown for some lightweight editing needs (like lecture notes) and being able to have page numbers would be great, no matter how bad they look.

**Comment by RoyiAvital** [comment]
+1
